File: ios/VisionCameraFaceDetector.m
#import <Foundation/Foundation.h>
#import <VisionCamera/FrameProcessorPlugin.h>
#import <VisionCamera/FrameProcessorPluginRegistry.h>
#import <VisionCamera/Frame.h>

#if defined __has_include && __has_include("VisionCameraFaceDetector-Swift.h")
#import "VisionCameraFaceDetector-Swift.h"
#else
#import <VisionCameraFaceDetector/VisionCameraFaceDetector-Swift.h>
#endif



@interface VisionCameraFaceDetector(FrameProcessorPluginLoader)
@end

@implementation VisionCameraFaceDetector(FrameProcessorPluginLoader)

+ (void)load {
    [FrameProcessorPluginRegistry addFrameProcessorPlugin:@"scanFaces"
                                          withInitializer:^FrameProcessorPlugin* _Nonnull(VisionCameraProxyHolder* _Nonnull proxy,
                                                                                          NSDictionary* _Nullable options) {
        return [[VisionCameraFaceDetector alloc] initWithProxy:proxy withOptions:options];
    }];
}

@end

File: ios/VisionCameraFaceDetector.swift
import VisionCamera

@objc(VisionCameraFaceDetector)
public class VisionCameraFaceDetector: FrameProcessorPlugin {
  public override init(proxy: VisionCameraProxyHolder, options: [AnyHashable: Any]! = [:]) {
    super.init(proxy: proxy, options: options)
  }

  public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable: Any]?) -> Any? {
    let buffer = frame.buffer
    let orientation = frame.orientation
    // code goes here
    return nil
  }
}
